src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -53,6 +55,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     if (userService.isUserLoggedIn()) {
       String logoutUrl = userService.createLogoutURL("/");
       loginStatus = new LoginStatus(true, logoutUrl);
+
+      // Create user entity if it doesn't already exist.
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      String userId = userService.getCurrentUser().getUserId();
+      Key userKey = KeyFactory.createKey("User", userId);
+      try {
+        datastore.get(userKey);
+      } catch (Exception e) {
```

Comment:
This should catch a more specific exception, as all exceptions do not mean that the user does not exist.

A different, possibly better, approach is to do a query (rather than a get) and check whether the count is 0. A query that returns an empty result should not throw an exception.

You can create an issue and add a TODO to fix in a future PR.